### ISSUE TITLE
Figma Transformer 

### ISSUE THREAD

**Issue by nikoloza** [symbo-ls/sketchql#2]
### tl;dr

We create components using props from a Design System setting. As the output, it transforms the object into a [DOMQL element](https://github.com/domql/domql/tree/feature/v2/packages/element) containing calculated properties for DOM, or React to render.

### Example:
```javascript
const Poster = {
  extends: [Link, Img],
  props: {
    boxSize: [140, 200],
    borderRadius: 12,
    padding: 16,
    color: 'black 0.95'
    background: 'oceanblue 0.35'
  }
}

DOM.create(Poster, document.body)
```
This will run through Symbols transformers and [create an element](https://github.com/domql/domql/tree/feature/v2/packages/element) with calculated CSS and other necessary properties. The [output](https://github.com/domql/domql/blob/main/test/element/create.test.js) looks like this:
```yaml
tag: "a"
attr: {href: '', target: '', aria-label: '', src: '', alt: '', …}
class:
  width: '140px',
  height: '200px',
  padding: '1em',
  color: 'rgba(0, 0, 0, 0.95)',
  backgroundColor: 'rgba(17, 128, 118, 0.35)',
  borderRadius: '0.825em',
  default: {boxSizing: 'border-box'}
transform: {emotion: 'smbls-roynbj smbls-m1cibc'}
```

---

1. To support it in Figma, first, we'll need to create a Figma transformer that will prepare all essential properties, and match them with Figma settings. Example transformers: 
- [transform-emotion](https://github.com/domql/domql/tree/feature/v2/packages/transform-emotion)
- [transform-react](https://github.com/domql/domql/tree/feature/v2/packages/transform-react)

2. Then, we can [create Figma components](https://www.figma.com/plugin-docs/api/properties/figma-createcomponent/#docsNav) using its API
